onebusaway-android/src/main/java/org/onebusaway/android/io/backup/RestorePreference.java

```diff
@@ -86,15 +87,23 @@ public void onClick(DialogInterface dialog, int which) {
     }
 
     void doRestore() {
-        Context context = getContext();
+        final Context context = getContext();
         ObaAnalytics.reportEventWithCategory(ObaAnalytics.ObaEventCategory.UI_ACTION.toString(),
                 context.getString(R.string.analytics_action_button_press),
                 context.getString(R.string.analytics_label_button_press_restore_preference));
         try {
             Backup.restore(context);
-            Toast.makeText(context,
-                    context.getString(R.string.preferences_db_restored),
-                    Toast.LENGTH_LONG).show();
+
+            ObaRegionsTask task = new ObaRegionsTask(context, new ObaRegionsTask.Callback() {
+                @Override
+                public void onRegionTaskFinished(boolean currentRegionChanged) {
+                    Toast.makeText(context,
```

Comment:
In the Toast example [here](https://developer.android.com/guide/topics/ui/notifiers/toasts.html), they use `getApplicationContext()` - I think that's a better option here as the Activity context could have been destroyed.

Also, instead of `context.getString(R.string.preferences_db_restored)`, I believe we can just pass in the resource ID:
`R.string.preferences_db_restored`

Sorry, I missed this in the first review.